code.py

```diff
@@ -251,13 +251,43 @@ def run_wifi():
 			mqtt.send_digirom_output(last_output)
 
 			while True:
-				mqtt.loop()
+				mqtt_loop()
 				if mqtt.get_subscribed_output(False) is not None:
 					break
 				if time.monotonic() - time_start >= 5:
 					break
 				time.sleep(0.1)
 
+mqtt_failure_start_time = time.monotonic()
+
+def mqtt_loop():
```

Comment:
How about moving all the failure counting into mqtt.loop(), giving that a return value, and just having the final error report here?
```
if not mqtt.loop():
	print("Maximum number of failures reached within 30 seconds. MQTT Failure...")
	connection_failure_alert("MQTT")
```